### PR TITLE
Standardize XCTest locations and harnesses

### PR DESCRIPTION
**Pull request by stuartmorgan**
- Moves XCTest files to the now-standard location
- Ensures that the harnesses are called RunnerTests for consistency
- Splits the image_picker unit tests out of the UI target into a new unit test target
- Moves existing google_sign_in tests into the harness, since they weren't being run. One new test, added since we accidentally stopped compiling the file, was removed since it crashed other tests in the suite (which has non-trivial global state, so fixing it wasn't feasible here; I've follow up on the PR that added the test).

## Pre-launch Checklist

- [x] I read the [Contributor Guide] and followed the process outlined there for submitting PRs.
- [x] I read the [Tree Hygiene] wiki page, which explains my responsibilities.
- [x] I read and followed the [Flutter Style Guide] and the [C++, Objective-C, Java style guides]. (Note that unlike the flutter/flutter repo, the flutter/plugins repo does use `dart format`. See [plugin_tool format])
- [x] I signed the [CLA].
- [ ] The title of the PR starts with the name of the plugin surrounded by square brackets, e.g. `[shared_preferences]`
- [ ] I listed at least one issue that this PR fixes in the description above.
- [ ] I updated pubspec.yaml with an appropriate new version according to the [pub versioning philosophy].
- [ ] I updated CHANGELOG.md to add a description of the change.
- [ ] I updated/added relevant documentation (doc comments with `///`).
- [ ] I added new tests to check the change I am making or feature I am adding, or Hixie said the PR is test exempt.
- [x] All existing and new tests are passing.

[Contributor Guide]: https://github.com/flutter/flutter/wiki/Tree-hygiene#overview
[Tree Hygiene]: https://github.com/flutter/flutter/wiki/Tree-hygiene
[Flutter Style Guide]: https://github.com/flutter/flutter/wiki/Style-guide-for-Flutter-repo
[C++, Objective-C, Java style guides]: https://github.com/flutter/engine/blob/master/CONTRIBUTING.md#style
[CLA]: https://cla.developers.google.com/
[flutter/tests]: https://github.com/flutter/tests
[breaking change policy]: https://github.com/flutter/flutter/wiki/Tree-hygiene#handling-breaking-changes
[Discord]: https://github.com/flutter/flutter/wiki/Chat
[pub versioning philosophy]: https://dart.dev/tools/pub/versioning
[plugin_tool format]: https://github.com/flutter/plugins/blob/master/script/tool/README.md#format-code

### CODE REVIEW

**Review by cyanglaz — APPROVED** [review]
LGTM

**Comment by stuartmorgan** [comment]
I'll look tomorrow to figure out what I broke.

File: packages/image_picker/image_picker/example/ios/Runner.xcodeproj/project.pbxproj [hunk 1]
```diff
@@ -514,6 +616,55 @@
...
+				CLANG_ANALYZER_NUMBER_OBJECT_CONVERSION = YES_AGGRESSIVE;
+				CLANG_CXX_LANGUAGE_STANDARD = "gnu++14";
+				CLANG_ENABLE_OBJC_WEAK = YES;
+				CLANG_WARN_DOCUMENTATION_COMMENTS = YES;
+				CLANG_WARN_QUOTED_INCLUDE_IN_FRAMEWORK_HEADER = YES;
+				CLANG_WARN_UNGUARDED_AVAILABILITY = YES_AGGRESSIVE;
+				CODE_SIGN_STYLE = Automatic;
+				GCC_C_LANGUAGE_STANDARD = gnu11;
+				INFOPLIST_FILE = RunnerTests/Info.plist;
+				IPHONEOS_DEPLOYMENT_TARGET = 14.5;
```
Comment (jmagman) [inline packages/image_picker/image_picker/example/ios/Runner.xcodeproj/project.pbxproj:633]: Don't override the project-level deployment target `IPHONEOS_DEPLOYMENT_TARGET` or `TARGETED_DEVICE_FAMILY`.  I usually also don't override `CLANG_*`, `GCC_*` or `MTL_*` but that's less important.

**Comment by jmagman** [comment]
> I'll look tomorrow to figure out what I broke.

> 	Double-quoted include "FlutterAppDelegate.h" in framework header, expected angle-bracketed instead
	Double-quoted include "FlutterMacros.h" in framework header, expected angle-bracketed instead
	Double-quoted include "FlutterPlugin.h" in framework header, expected angle-bracketed instead
	Double-quoted include "FlutterBinaryMessenger.h" in framework header, expected angle-bracketed instead
	Double-quoted include "FlutterMacros.h" in framework header, expected angle-bracketed instead
	Double-quoted include "FlutterChannels.h" in framework header, expected angle-bracketed instead
	Double-quoted include "FlutterBinaryMessenger.h" in framework header, expected angle-bracketed instead
	Double-quoted include "FlutterCodecs.h" in framework header, expected angle-bracketed instead
	Double-quoted include "FlutterMacros.h" in framework header, expected angle-bracketed instead
	Double-quoted include "FlutterCodecs.h" in framework header, expected angle-bracketed instead
	Double-quoted include "FlutterPlatformViews.h" in framework header, expected angle-bracketed instead
	Double-quoted include "FlutterCodecs.h" in framework header, expected angle-bracketed instead
	Double-quoted include "FlutterMacros.h" in framework header, expected angle-bracketed instead
	Double-quoted include "FlutterTexture.h" in framework header, expected angle-bracketed instead
	Double-quoted include "FlutterMacros.h" in framework header, expected angle-bracketed instead
	Double-quoted include "FlutterBinaryMessenger.h" in framework header, expected angle-bracketed instead
	Double-quoted include "FlutterCallbackCache.h" in framework header, expected angle-bracketed instead
	Double-quoted include "FlutterMacros.h" in framework header, expected angle-bracketed instead
	Double-quoted include "FlutterChannels.h" in framework header, expected angle-bracketed instead
	Too many errors emitted, stopping now
	Could not build module 'Flutter'
	Could not build module 'image_picker'
	Testing cancelled because the build failed.

To import the Flutter framework, the new `image_picker` `RunnerTests` target shouldn't set `CLANG_WARN_QUOTED_INCLUDE_IN_FRAMEWORK_HEADER=YES`

File: packages/image_picker/image_picker/example/ios/Runner.xcodeproj/project.pbxproj (hunk 1 above)
Comment (stuartmorgan) [reply packages/image_picker/image_picker/example/ios/Runner.xcodeproj/project.pbxproj]: I forgot Xcode generates a bunch of target-level settings by default. Fixed, and I'll go add that to the docs.

**Review by jmagman — APPROVED** [review]
LGTM

**Comment by stuartmorgan** [comment]
I'm not sure what I still have wrong in the new webview test target; I didn't see anything obvious comparing against one of the working projects.

**Comment by stuartmorgan** [comment]
🤦🏻 Log reading fail on my part. Somehow I messed up the target membership of the test files in the webview test.

**Pull request merged**